src/fdc3/dotnet/DesktopAgent.Client/src/MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client/Infrastructure/Internal/IntentsClient.cs

```diff
@@ -78,4 +78,41 @@ public async ValueTask<IAppIntent> FindIntentAsync(string intent, IContext? cont
 
         return response.AppIntent;
     }
+
+    public async ValueTask<IEnumerable<IAppIntent>> FindIntentsByContextAsync(IContext context, string? resultType = null)
+    {
+        var request = new FindIntentsByContextRequest
+        {
+            Fdc3InstanceId = _instanceId,
+            Context = JsonSerializer.Serialize(context, _jsonSerializerOptions),
+            ResultType = resultType
+        };
+
+        if (_logger.IsEnabled(LogLevel.Debug))
+        {
+            _logger.LogDebug("Finding intents by context. Context Type: {ContextType}, Result Type: {ResultType}", context.Type, resultType);
+        }
+
+        var response = await _messaging.InvokeJsonServiceAsync<FindIntentsByContextRequest, FindIntentsByContextResponse>(
+            Fdc3Topic.FindIntentsByContext,
+            request,
+            _jsonSerializerOptions);
+
+        if (response == null)
+        {
+            throw ThrowHelper.MissingResponse();
+        }
+
+        if (!string.IsNullOrEmpty(response.Error))
+        {
+            throw ThrowHelper.ErrorResponseReceived(response.Error);
+        }
+
+        if (response.AppIntents == null)
+        {
+            throw ThrowHelper.AppIntentIsNotDefined("null", context.Type, resultType);
```

Comment:
Similarly to before, it's strange to me to throw on a null AppIntents list, but not on an empty array.

Reply:
I would keep it as it is as the backend can send back an empty `AppIntent` list. I would not expect the client to throw an exception when it just returns an empty list.